tests/gtest/aviflosslesstest.cc

```diff
@@ -64,26 +62,70 @@ void ReadImageSimple(const std::string& file_name, avifPixelFormat pixel_format,
   ASSERT_NE(file_format, AVIF_APP_FILE_FORMAT_UNKNOWN);
 }
 
+// Checks whether an image is grayscale. ASSERT_NO_FATAL_FAILURE should be used
+// when calling it.
+void IsGray(const std::string& path, bool& is_gray) {
+  ImagePtr image(avifImageCreateEmpty());
+  ASSERT_NE(image, nullptr);
+  image->matrixCoefficients = AVIF_MATRIX_COEFFICIENTS_UNSPECIFIED;
+  ASSERT_NE(
+      AVIF_APP_FILE_FORMAT_UNKNOWN,
+      avifReadImage(path.c_str(), AVIF_PIXEL_FORMAT_NONE, /*requestedDepth=*/0,
+                    /*chromaDownsampling=*/AVIF_CHROMA_DOWNSAMPLING_AUTOMATIC,
+                    /*ignoreColorProfile=*/true, /*ignoreExif=*/true,
+                    /*ignoreXMP=*/true,
+                    /*allowChangingCicp=*/true, /*ignoreGainMap=*/true,
+                    AVIF_DEFAULT_IMAGE_SIZE_LIMIT, image.get(),
+                    /*outDepth=*/nullptr, /*sourceTiming=*/nullptr,
+                    /*frameIter=*/nullptr));
+  is_gray = image->yuvFormat == AVIF_PIXEL_FORMAT_YUV400;
+}
+
 // The test parameters are: the file path, the matrix coefficients and the pixel
 // format.
-class LosslessTest
+class EncodeDecodeMemory
     : public testing::TestWithParam<std::tuple<std::string, int, int>> {};
 
-// Tests encode/decode round trips.
-TEST_P(LosslessTest, EncodeDecode) {
-  const std::string& file_name = std::get<0>(GetParam());
+// Tests encode/decode round trips in RAM.
+TEST_P(EncodeDecodeMemory, RoundTrip) {
+  const std::string& file_path =
+      std::string(data_path) + std::get<0>(GetParam());
   const avifMatrixCoefficients matrix_coefficients =
       static_cast<avifMatrixCoefficients>(std::get<1>(GetParam()));
   const avifPixelFormat pixel_format =
       static_cast<avifPixelFormat>(std::get<2>(GetParam()));
-  // Ignore ICC when going from RGB to gray.
-  const avifBool ignore_icc =
-      pixel_format == AVIF_PIXEL_FORMAT_YUV400 ? AVIF_TRUE : AVIF_FALSE;
 
-  // Read a ground truth image but ask for certain matrix coefficients.
+  // Check if the input image is grayscale.
+  bool gt_is_gray = false;
```

Comment:
`is_gray` is probably fine here since there's no external parameter you're trying to distinguish it from.